### DISCUSSION TITLE
Content question for Caroline - "Download" or "Export"

### DISCUSSION THREAD

**Discussion by johannaweintraub** [twilio-labs/paste#990] [Q&A] [answered]
HI @ctecks! We were looking at some tables in our Crit pod and saw that we have inconsistency around using the word "Download" vs "Export". Is there guidance on which term we should use? The functionality is for downloading/exporting table data in CSV and sometimes other formats. Thanks!

Here is an example: https://share.goabstract.com/bf56d29b-c928-4c50-b5f8-e6bca8724a75?collectionLayerId=d87c52b2-1d11-4f45-96b0-6a64c7549810&mode=design

**Comment by ctecks** [comment]
Ooh, interesting question. I looked over this page with Sang last week - as you say the user can export the data in the file in different formats, and there's a second step after they hit the button. 

In this case, as the user can choose the format they want, stick with 'Export' - the content needs to be converted before the item is locally stored. 

If it's a straight-up, one-click, local save of an item in a set format (graph, image, file, log etc.), and the only step that follows is a system dialogue to choose a download destination, stick with 'Download'.

How does that strike you as workable guidance in terms of logic and applicability across the platform?

**Comment by johannaweintraub** [comment]
Thanks so much @ctecks. Ideally, I would think having a single term might be better for the sake of building a consistent experience. Is there a reason why you think it makes sense to use a different term based on whether it is one-click or 2-clicks?

**Reply by sang-yeo** [reply]
Here is the new [link](https://share.goabstract.com/bf56d29b-c928-4c50-b5f8-e6bca8724a75?collectionLayerId=5e86732e-04a4-4a52-bc87-a86fa76553fe&mode=design) for this thread I updated the sketch art board and it won't work anymore.

**Reply by ctecks** [reply]
Sure @johannaweintraub, it's for the reason of consistency we should have two specific terms - so that we can clearly and consistently signal to the user what the button does. 

What we're proposing is that 'Download' = local save of file in a set format, while 'Export' = conversion of file to a range of formats, then local save. Those feel like quite different propositions to the user - one enables them to manipulate the file, the other does not - as such I'm advocating for specific terms to describe them so the user can understand what to expect. Per Paste guidance, button actions should be specific. 

Imagine that we use one term for all 'local save' actions - one experience is one-click, while the other involves selecting a file format. I, the user, select something that I expect to be one-click, but that turns out to be more intricate with a decision to make - on the other hand, I select something I expect to be able to convert to my desired file format, but it saves locally without giving me the option. Both experiences are potentially jarring - equally, I don't know at a glance what I can do with an item on a page (e.g. can I convert the format of this item or not?) Yes, the user is seeking the same goal in both instances, but the two routes for the user to get there are two different experiences. I'd liken it to the difference between Save and Save As, if you will. 

Shopify lay it out in [their style guide](https://polaris.shopify.com/content/actionable-language#section-export-vs-download) very neatly - 100% going to borrow some of their formatting. Let me know if you want to chat further.

**Comment by johannaweintraub** [comment]
Thanks so much for the detailed response. Really appreciate it. I just had a comically long conversation with an engineer about this to try to better understand the technical side of this. I'll lay out what I learned. 

In all our cases, whether you have only a CSV option or multiple file format options, the data needs to be coverted from the database into the given format before being downloaded to the computer. IOW, the download to CSV case still has to convert the data into the CSV format. In Sang's case, you are doing the exact same thing but you have more than one file format option. So, in essence, in all our cases we are doing both the conversion and the downloading.

Export is a fairly general term that generally means to both convert the data into a new format and then to move it to somewhere else (could be as a file on your computer or another database like Airtable). 

Download is a more specific term for moving data from the database to the computer.

In Shopify's case, I imagine they have a lot of integrations where you're exporting data from their system into another system. In our cases, we're downloading the data to the user's computer. Although I could imagine having other functionality in the future where a user could export the data into another system, llike Airtable.

Whether or not the user gets the dialog to choose where to save it to or whether the browser automatically saves it is dependent on the browser settings.

I'm happy to set up some time during office hours to discuss this further, but I wanted to document this here for the sake of writing it down. Let me know your thoughts when you have a moment.

**Reply by corinne-nagel** [reply]
Hi @johannaweintraub could you please share what the consensus was for this so we all have it to reference? Thanks!

**Comment by johannaweintraub** [comment]
Thanks for following up. I've been continuing to research this pattern and
collect feedback from various groups. I've come up with 2 potential
heuristics we can use to establish a pattern. This will require testing to
see what resonates best with our customers. After that, I will contribute
this as a pattern.

Here are the 2 possible directions:
https://share.goabstract.com/d81de498-a66d-4a6e-acf7-6c8c7eda2be4

**Reply by corinne-nagel** [reply]
Sounds good, thanks for the update!

**Reply by corinne-nagel** [reply]
Update: more testing is needed and will be updated end of Feb. Doc is started here for reference:
https://docs.google.com/document/d/1jdRSlHC0_cOoQ37ffesK72WiNTl3d0zhsbDMGy93yzA/edit

**Comment by gloriliale** [comment, marked as answer]
To wrap this up, we will be using "export" and will shortly be adding a data handling pattern for this 🎉
